contract_manager/src/governance.ts

```diff
@@ -206,26 +236,32 @@ export class WormholeMultiSigTransaction {
 
   async getState() {
     const proposal = await this.squad.getTransaction(this.address);
-    return proposal.status; //TODO: make it typed and parse the status to a more readable format
+    // Converts the status object to a string e.g
+    // { "active":{} } => "active"
+    return Object.keys(proposal.status)[0];
   }
 
-  async execute(): Promise<SubmittedWormholeMessage> {
+  async execute(): Promise<SubmittedWormholeMessage[]> {
     const proposal = await this.squad.getTransaction(this.address);
     const signatures = await executeProposal(
       proposal,
       this.squad,
       this.cluster
     );
+    const msgs: SubmittedWormholeMessage[] = [];
     for (const signature of signatures) {
       try {
-        return SubmittedWormholeMessage.fromTransactionSignature(
-          signature,
-          this.cluster
+        msgs.push(
+          await SubmittedWormholeMessage.fromTransactionSignature(
+            signature,
+            this.cluster
+          )
         );
       } catch (e: any) {
         if (!(e instanceof InvalidTransactionError)) throw e;
       }
     }
+    if (msgs.length > 0) return msgs;
```

Comment:
you sure about this check? seems like if none of the txs sent wormhole messages, then the proposal executes successfully and returns 0 messages

Reply:
So, this class was mainly designed for Wormhole messages, so I thought it should throw an exception if there is actually no wormhole instructions inside. I can add a check in constructor to verify this fact, but if someone is manually creating this object, hopefully they know what they are doing. I will add some comments. In the end, I think the main usage of this class will be in manual/dev testing as a cranker would do this automatically on our pre-configured vaults.